src/basic/mount-util.c

```diff
@@ -689,3 +689,35 @@ int umount_verbose(const char *what) {
                 return log_error_errno(errno, "Failed to unmount %s: %m", what);
         return 0;
 }
+
+const char *mount_propagation_flags_to_string(unsigned long flags) {
+
+        switch (flags & (MS_SHARED|MS_SLAVE|MS_PRIVATE)) {
+
+        case MS_SHARED:
+                return "shared";
+
+        case MS_SLAVE:
+                return "slave";
+
+        case MS_PRIVATE:
+                return "private";
+        }
+
+        return NULL;
+}
+
+unsigned long mount_propagation_flags_from_string(const char *name) {
+
+        if (isempty(name))
```

Comment:
Well, most of our xyz_from_string() functions check for NULL (since the `DEFINE_STRING_TABLE_LOOKUP` macro does that), and hence I figure we should check that here too.

Note that the function is called mount_propagation_flags_from_string() and not just mount_flags_from_string(), it focusses only on the three propagation bits, and always masks the rest out. And the three propagation flags are actually more like an enum: you pick one, you never combine them. As such I really wanted to provide this with the "feel" of an enum translator call... (except that we use "0" as "can't translate" instead of "-1", because -1 might mean weird stuff, and "0" actually means "no propagation flag set" which is what we want..)